From f91d7f5fa5cbc91f480e4eee2c5e1ce4dd396b6f Mon Sep 17 00:00:00 2001
From: Kiveisha Yevgeniy <yevgeniy.kiveisha@intel.com>
Date: Sun, 11 Jan 2015 19:14:14 +0200
Subject: [PATCH 15/28] camera ov5693: device power_up retries.

Added __power_up function which called from power_up and
incase of failure it will be called again.
Number or retries limited to 4. Added to both cameras.

Issue: GMINL-3806
Change-Id: Ib300182038476fada3c1fb504d16cddb766254a6
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7550
Signed-off-by: Kiveisha Yevgeniy <yevgeniy.kiveisha@intel.com>
Signed-off-by: Per Persson <per.persson@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/83124
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/440501
---
 .../camera/drivers/media/i2c/ov5693-ecs/ov5693.c       | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)
 mode change 100755 => 100644 drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
old mode 100755
new mode 100644
index 0c93e9f..2135828
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1384,8 +1384,7 @@ static int gpio_ctrl(struct v4l2_subdev *sd, bool flag)
 	return ret;
 }
 
-
-static int power_up(struct v4l2_subdev *sd)
+static int __power_up(struct v4l2_subdev *sd)
 {
 	struct ov5693_device *dev = to_ov5693_sensor(sd);
 	struct i2c_client *client = v4l2_get_subdevdata(sd);
@@ -1465,6 +1464,21 @@ static int power_down(struct v4l2_subdev *sd)
 	return ret;
 }
 
+static int power_up(struct v4l2_subdev *sd)
+{
+	static const int retry_count = 4;
+	int i, ret;
+
+	for (i = 0; i < retry_count; i++) {
+		ret = __power_up(sd);
+		if (!ret)
+			return 0;
+
+		power_down(sd);
+	}
+	return ret;
+}
+
 static int ov5693_s_power(struct v4l2_subdev *sd, int on)
 {
 	int ret;
-- 
1.9.1

